### DISCUSSION TITLE
Lesson 7 running test on fork mainnet or testnet

### DISCUSSION THREAD

**Discussion by miladad8** [Cyfrin/foundry-full-course-cu#1161] [General]
Hi. I am having a hard time understanding lesson 7 tests when we use this command  `forge test --fork-url $MAINNET_RPC_URL`. 

shouldn't we spend some real ETH for testing our scripts? Or even on sepolia testnet  spend some sepolia ETH?

sorry if it's a dummy question.

**Comment by ghost** [comment]
When you fork mainnet to test your contract, you're grabbing the current state of mainnet and use it on your local blockchain (anvil) to work with existing deployed contracts and/or use data that is not available on your local blockchain.

Basically you're just running a more "realistic" simulation of your contract using real data.

If you use an account with no ETH and use it in `prank` you will see that the test will not revert because of lacks of funds.

**Reply by miladad8** [reply]
thank you so much, my friend!
